### PR TITLE
[INFINITY-1778] Insulate sdk_install from failures

### PR DESCRIPTION
**Pull request by benclarkwood**
Retry install and uninstall 5 times, waiting 5 seconds between attempts. Insulates against Marathon flakes.

### CODE REVIEW

File: testing/sdk_utils.py [hunk 1]
```diff
@@ -32,3 +33,15 @@ def gc_frameworks():
...
+
+
+def try_throws_n_times(predicate, attempts=1, sleep_seconds=0):
+    for index in range(attempts):
+        try:
+            return predicate()
+        except:
+            if index == attempts-1:
+                raise
+            out("Encountered an exception: {}".format(traceback.format_exc()))
```
Comment (jsrodman) [inline testing/sdk_utils.py]: Do you really want this after the raise?  Sort of odd / confusing.

Comment (benclarkwood) [inline testing/sdk_utils.py]: if attempts = 3, then:

1: log exception, continue
2: log exception, continue
3: raise, don't log it

File: testing/sdk_install.py [hunk 2]
```diff
@@ -56,6 +68,11 @@ def install(
 
 
 def uninstall(service_name, package_name=None, role=None, principal=None, zk=None):
+    sdk_utils.try_throws_n_times(lambda: _uninstall(service_name, package_name, role, principal, zk),
```
Comment (jsrodman) [inline testing/sdk_install.py]: I'd probably have just done
```python
     def _uninstall(service_name=service=name, package_name=package_name...):
        # uninstall work
    sdk_utils.try_throws_n_times(_uninstall, 5, 5)
```

Comment (benclarkwood) [inline testing/sdk_install.py]: I did not know Python did that. Seems magical

**Comment by benclarkwood** [comment]
retest this please

**Comment by benclarkwood** [comment]
retest this please

**Comment by benclarkwood** [comment]
retest this please

File: testing/sdk_utils.py (hunk 1 above)
Comment (jsrodman) [inline testing/sdk_utils.py]: I was suggesting I'd log it anyway. No guarantees all the exceptions are the same.

**Comment by benclarkwood** [comment]
retest this please

**Comment by benclarkwood** [comment]
retest this please (I merged elastic test fixes in PR #1121)

**Comment by benclarkwood** [comment]
Yep. Good idea.

**Comment by benclarkwood** [comment]
TBH Given this looks like it's about to pass, I'm going to do that in a separate PR.

**Comment by nickbp** [comment]
:+1:

**Pull request merged**